frontend/packages/operator-lifecycle-manager/src/components/operator-install-page.tsx

```diff
@@ -118,28 +115,33 @@ const InstallNeedsApprovalMessage: React.FC<InstallNeedsApprovalMessageProps> =
   </>
 );
 
-const CreateCRDButton: React.FC<CRDButtonProps> = ({ obj, disabled }) => {
-  if (!getCRDFromObject(obj)) {
+export const CreateInitializationResourceButton: React.FC<InitializationResourceButtonProps> = ({
+  obj,
+  disabled,
+  targetNamespace,
+}) => {
+  if (!getInitializationResourceJSON(obj)) {
     return null;
   }
-  let cr = null;
+  let initializationResource = null;
   try {
-    cr = JSON.parse(getCRDFromObject(obj));
+    initializationResource = JSON.parse(getInitializationResourceJSON(obj));
   } catch (error) {
     const errorMsg = error.message;
     errorModal({ errorMsg });
   }
-  const crNamespace = cr?.[0]?.metadata.namespace;
-  const apiVersion = cr?.[0]?.apiVersion;
-  const kind = cr?.[0]?.kind;
-  const { group, version } = groupVersionFor(apiVersion);
-  const reference = referenceForGroupVersionKind(group)(version)(kind);
+  const reference = referenceFor(initializationResource);
+  const model = modelFor(reference);
+  const initializationResourceNamespace = model?.namespaced
+    ? initializationResource?.metadata?.namespace || targetNamespace
+    : null;
+  const kind = initializationResource?.kind;
   return (
     <Link
       to={`${resourcePathFromModel(
         ClusterServiceVersionModel,
         obj.metadata.name,
-        crNamespace,
+        initializationResourceNamespace,
```

Comment:
This isn't quite right if the initialization resource is from a different namespace than the CSV. The operand details form won't work in that case. We probably need to link to a YAML editor.

@tlwu2013 FYI

Reply:
It sounds like I'm mistaken and that this works.